File: Java/dependency-injection/bastard-injection/src/main/java/com/github/grzesiek_galezowski/DependencyInjection/_1_ControlFreak/Core/TeleComSystem.cs
using DependencyInjectionBefore._1_ControlFreak.Inbound;
using DependencyInjectionBefore._1_ControlFreak.Outbound;

namespace DependencyInjectionBefore._1_ControlFreak.Core
{
  class TeleComSystem
  {
    private readonly AcmeProcessingWorkflow _processingWorkflow;
    private readonly BinaryUdpInbound _inbound;
    private readonly XmlTcpOutbound _outbound;

    public TeleComSystem()
    {
      _inbound = new BinaryUdpInbound();
      _outbound = new XmlTcpOutbound();
      _processingWorkflow = new AcmeProcessingWorkflow();
    }

    public void Start()
    {
      _inbound.SetDomainLogic(_processingWorkflow);
      _processingWorkflow.SetOutbound(_outbound);
      _inbound.StartListening();
    }
  }
}

File: Java/dependency-injection/control-freak/src/main/java/com/github/grzesiek_galezowski/DependencyInjection/_1_ControlFreak/Core/TeleComSystem.java
package com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak.Core;

import com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak.Inbound.BinaryUdpInbound;
import com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak.Outbound.XmlTcpOutbound;

public class TeleComSystem {
    private final AcmeProcessingWorkflow _processingWorkflow;
    private final BinaryUdpInbound _inbound;
    private final XmlTcpOutbound _outbound;

    public TeleComSystem() {
      _inbound = new BinaryUdpInbound();
      _outbound = new XmlTcpOutbound();
      _processingWorkflow = new AcmeProcessingWorkflow();
    }

    public void start() {
      _inbound.setDomainLogic(_processingWorkflow);
      _processingWorkflow.setOutbound(_outbound);
      _inbound.startListening();
    }

  }